runtime-sofa-boot-starter/src/main/java/com/alipay/sofa/runtime/spring/ServiceBeanFactoryPostProcessor.java

```diff
@@ -123,43 +126,75 @@ private void generateSofaServiceDefinitionOnMethod(String beanId,
                                                        ConfigurableListableBeanFactory beanFactory) {
         Class<?> returnType;
         Class<?> declaringClass;
-        Method method = null;
+        List<Method> candidateMethods = new ArrayList<>();
 
         MethodMetadata methodMetadata = beanDefinition.getFactoryMethodMetadata();
         try {
             returnType = ClassUtils.forName(methodMetadata.getReturnTypeName(), null);
             declaringClass = ClassUtils.forName(methodMetadata.getDeclaringClassName(), null);
+        } catch (Throwable throwable) {
+            // it's impossible to catch throwable here
+            SofaLogger.error(throwable,
+                "Failed to parse factoryBeanMethod of BeanDefinition( {0} )", beanId);
+            return;
+        }
+        if (methodMetadata instanceof StandardMethodMetadata) {
+           candidateMethods.add(((StandardMethodMetadata) methodMetadata).getIntrospectedMethod());
+        } else {
             for (Method m : declaringClass.getDeclaredMethods()) {
+                // check methodName and return type
+                if (!m.getName().equals(methodMetadata.getMethodName())
+                        || !m.getReturnType().getTypeName().equals(methodMetadata.getReturnTypeName())) {
+                    continue;
+                }
+
+                // check bean method
+                if (!AnnotatedElementUtils.hasAnnotation(m, Bean.class)) {
+                    continue;
+                }
 
                 Bean bean = m.getAnnotation(Bean.class);
```

Comment:
`@Bean` could be a meta-annotation, which causes an NPE to be thrown.

emmm... but who will do that? just ignore me.